### ISSUE TITLE
Propel is leaking in Global Scope

### ISSUE THREAD

**Issue by gauntface** [GoogleChrome/Propel#46]
TL;DR Propel is leaking a tonne of variables on the global scope and Mocha wasn't catching it due to the formation of the tests

Mocha has leak detection, but the way it works is it compares the current state *before* the test to the state *after* the test.

The problem with this is that the mocha test page loads the script in the html (i.e. before the tests run)

    <script src="/dist/client.js"></script>

This means the leaks aren't treated as leaks. The fix for the test is to add a test that asynchronously loads the script and this will trigger the leak detection.

      it('should load window.goog.propel.Client without leaks', done => {
        // By leaks this is referring to the only thing Propel
        // should add to the global scope (i.e. window) is goog
        const scriptElement = document.createElement('script');
        scriptElement.setAttribute('type', 'text/javascript');
        scriptElement.src = '/dist/client.js';
        document.querySelector('head').appendChild(scriptElement)
        .onload = () => {
          console.log('Script Loaded');
          done();
        };
      });

**Comment by wibblymat** [comment]
Hmm, I thought that in an ES2015 module you got a separate 'module global' scope, much like in a node module. Did I misunderstand, or is there a limitation of Babel that's causing this?

**Comment by gauntface** [comment]
Babel config sounds like the thing
